### PR TITLE
Migration React Native SDK To Kustomer Chat 2.0

### PR DESCRIPTION
**Pull request by pauleta6**

### CODE REVIEW

File: android/src/main/java/co/reby/rnkustomersdk/KustomerSDKModule.kt [hunk 1]
```diff
@@ -0,0 +1,81 @@
...
+    private fun toMap(readableMap: ReadableMap): Map<String, Any> {
+        val map: MutableMap<String, Any> = HashMap()
+        val iterator: ReadableMapKeySetIterator = readableMap.keySetIterator()
+    
+        while (iterator.hasNextKey()) {
+            val key: String = iterator.nextKey()
+            val type: ReadableType = readableMap.getType(key)
+
+            when (type) {
+                ReadableType.Boolean -> map.put(key, readableMap.getBoolean(key))
```
Comment (sejoDjoseon) [inline android/src/main/java/co/reby/rnkustomersdk/KustomerSDKModule.kt]: put is deprecated: ```map[key] = readableMap.getBoolean(key).toString()```

File: android/src/main/java/co/reby/rnkustomersdk/KustomerSDKModule.kt [hunk 2]
```diff
@@ -0,0 +1,81 @@
...
+                custom = toMap(custom!!)
+        )
+
+        runBlocking {
+            Kustomer.getInstance().describeCustomer(attributes)
+        }
+    }
+
+    private fun toMap(readableMap: ReadableMap): Map<String, Any> {
+        val map: MutableMap<String, Any> = HashMap()
```
Comment (sejoDjoseon) [inline android/src/main/java/co/reby/rnkustomersdk/KustomerSDKModule.kt]:
```
    private fun toMap(readableMap: ReadableMap): Map<String, String> {
        val map: MutableMap<String, String> = HashMap()
```

File: ios/KustomerSDK.swift [hunk 3]
```diff
@@ -0,0 +1,76 @@
...
+
+  @objc(identify:)
+  func identify(_ hash: NSString, resolver resolve: @escaping RCTPromiseResolveBlock,
+                          rejecter reject: @escaping RCTPromiseRejectBlock) -> Void{
+
+         if(Kustomer.chatProvider.currentCustomer() == nil){
+             Kustomer.logIn(jwt: hash as String){ result in
+                 switch result {
+                     case .success:
+                     resolve("success")
```
Comment (sejoDjoseon) [inline ios/KustomerSDK.swift]: in android returns a true, should not be success(true)?

**Review by sejoDjoseon — COMMENTED** [review]
index.d.ts is missing for support in typescript projects

**Review by FerranReby — APPROVED** [review]
Fix the comments and thats it!!! Great work @pauleta6 😊 ☺️

File: android/src/main/java/co/reby/rnkustomersdk/KustomerSDKModule.kt (hunk 1 above)
Comment (pauleta6) [reply android/src/main/java/co/reby/rnkustomersdk/KustomerSDKModule.kt]: All done

File: index.d.ts [hunk 4]
```diff
@@ -0,0 +1,14 @@
+declare module "react-native-kustomer-sdk" {
```
Comment (sejoDjoseon) [inline index.d.ts:1]: ``@rebythecode/react-native-kustomer-sdk``

File: package.json [hunk 5]
```diff
@@ -1,7 +1,7 @@
 {
   "name": "react-native-kustomer-sdk",
```
Comment (sejoDjoseon) [inline package.json:2]: ```@rebythecode/react-native-kustomer-sdk```

**Review by sejoDjoseon — APPROVED** [review]

File: index.d.ts (hunk 4 above)
Comment (pauleta6) [reply index.d.ts:1]: Not anymore

**Pull request merged**